src/main/java/de/tum/in/www1/artemis/web/rest/UserJWTController.java

```diff
@@ -115,6 +133,43 @@ public ResponseEntity<JWTToken> authorizeSAML2(@RequestBody final String body) {
         return new ResponseEntity<>(new JWTToken(jwt), httpHeaders, HttpStatus.OK);
     }
 
+    /**
+     * Generates a personal access token (JWT token) for the current user.
+     *
+     * @param lifetimeMilliseconds the lifetime for the JWT token
+     * @return a JWT Token
+     */
+    @PostMapping("/personal-access-token")
+    @FeatureToggle(Feature.PersonalAccessTokens)
+    public ResponseEntity<JWTToken> getPersonalAccessToken(@RequestBody Long lifetimeMilliseconds) {
+        User user = userRepository.getUser();
+
+        if (!user.getActivated()) {
+            throw new UserNotActivatedException("User was disabled!");
+        }
+
+        long difference = this.personalAccessTokenMaxLifetimeMilliseconds - lifetimeMilliseconds;
```

Comment:
Since this variable is only needed inside the if-statement, I'd move it there as well